test/test_multi.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+python test_multi.py &
+wait
```

Comment:
@voznesenskym Do you know how to incorporate such a test into the CI?

Reply:
Multi process testing is tricky. You'd have to do a lot of mocking, and ensure that it reliably fails 100% of the time without this fix, or the flaky nature of the test won't give us the protection we want from a test.

Reply:
@voznesenskym I agree with your point that it might not provide 100% guarantee of protection but if the UT is created from the real usages, like multi-process inference, it would provide a good level of confidence that we are good to support such usages. On the other hand, if we already have such multi-process tests in the integration tests, we probably don't have to add a similar UT here. I'm open for better suggestions.

Reply:
@Valentine233 If we decide to add such a test, please make sure the test has good result checkers, e.g. the compiled function of each process should return expected result, the cache is cleared before tests etc.

Reply:
@voznesenskym Any comments on the test? How about we add multi-instance inference tests into the integration test later and remove such UTs here? We can add an issue to track this.

Reply:
I don't have a problem with the idea of testing - I do not know how we can make a test that fails 100% of the time without this fix. If we have a test that can give false negatives (regression, but test still passes) - that is not a test worth having.